multimodal/src/autogluon/multimodal/optimization/lit_distiller.py

```diff
@@ -154,15 +171,34 @@ def __init__(
         self.temperature = temperature
         self.hard_label_weight = hard_label_weight
         self.soft_label_weight = soft_label_weight
+        self.intermediate_loss_weight = intermediate_loss_weight
+        self.rkd_distance_loss_weight = rkd_distance_loss_weight
+        self.rkd_angle_loss_weight = rkd_angle_loss_weight
         self.hard_label_loss_func = hard_label_loss_func
         self.soft_label_loss_func = soft_label_loss_func
+        self.intermediate_loss_func = intermediate_loss_func
         if isinstance(validation_metric, BaseAggregator) and custom_metric_func is None:
             raise ValueError(
                 f"validation_metric {validation_metric} is an aggregation metric,"
                 "which must be used with a customized metric function."
             )
         self.custom_metric_func = custom_metric_func
 
+        # Adapt student's intermediate feature to teacher's
+        if isinstance(self.teacher_model, MultimodalFusionMLP) or isinstance(self.teacher_model, MultimodalFusionTransformer):
+            pass # skip fusion model for now
```

Comment:
We can add attributes in `MultimodalFusionMLP ` and `MultimodalFusionTransformer `:

`self.out_features =  in_features`

In this way, we can also support distillation for the fusion models.

Reply:
Shall we explicitly align each models in fusion between teacher and student or we can assume they have the same amount and same order, self.model = [hftext, timm, categorical_mlp] for both teacher and student?

Reply:
Aligning each individual backbone is ok when the teacher and student have the same late-fusion structure. But if teacher has hf_text and timm_image but student only has hf_text, then it may not be good. In this case, we expect the student hf_text output can inherit information from both the teacher's hf_text and timm_image, instead of only teachers's hf_text.

Reply:
So, the better choice is we only distill the final output feature. No need consider whether teacher and student are late-fusion or not.

Reply:
Is it possible for the student's text encoder to learn the feature of teacher's image/numerical encoder? It would be a lot easier if we only distill the final output though.

Reply:
I think we can do this with new config supports. But it seems that we don't such needs currently.

Reply:
Sure, I'll distill the fusion model output feature for now. And we can have a test when we have a good multimodal example to work on.

Reply:
Done